ambry-store/src/test/java/com.github.ambry.store/CompactionManagerTest.java

```diff
@@ -0,0 +1,291 @@
+/**
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.store;
+
+import com.codahale.metrics.MetricRegistry;
+import com.github.ambry.config.StoreConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.utils.MockTime;
+import com.github.ambry.utils.TestUtils;
+import com.github.ambry.utils.Time;
+import com.github.ambry.utils.UtilsTest;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.ScheduledExecutorService;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+
+/**
+ * Unit tests {@link CompactionManager}.
+ */
+public class CompactionManagerTest {
+
+  private static long CAPACITY_IN_BYTES = 10 * 1024 * 1024;
+  // the properties that will used to generate a StoreConfig. Clear before use if required.
+  private final Properties properties = new Properties();
+  private final Time time = new MockTime();
+  private MockBlobStore blobStore;
+  private CompactionManager compactionManager;
+  private long messageRetentionTimeInMs;
+
+  /**
+   * Instantiates {@link CompactionManagerTest} with the required cast
+   * @throws InterruptedException
+   */
+  public CompactionManagerTest() throws InterruptedException {
+    StoreConfig config = new StoreConfig(new VerifiableProperties(properties));
+    messageRetentionTimeInMs = config.storeDeletedMessageRetentionDays * Time.SecsPerDay * Time.MsPerSec;
+    time.sleep(2 * messageRetentionTimeInMs);
+    MetricRegistry metricRegistry = new MetricRegistry();
+    StorageManagerMetrics metrics = new StorageManagerMetrics(metricRegistry);
+    blobStore = new MockBlobStore("", config, null, null, metrics, null, 0, null, null, null, time);
```

Comment:
if most of these are `null` or empty string why have them as arguments to the constructor of `MockBlobStore`? Can just set them to `null` (or empty string) when calling the super constructor?